### DISCUSSION TITLE
Raspotify crashes when USB DAC disconnects

### DISCUSSION THREAD

**Discussion by wgorski** [dtcooper/raspotify#599] [Q&A] [answered]
I have a raspotify server running which outputs sound to an external USB DAC. Whenever the DAC is turned off or the input is changed to something other than USB (like Toslink), raspotify crashes. Following the crash, it's not possible to connect to it for some time.

It would be great if raspotify would keep streaming the music even though no DAC is connected. This way, when the DAC reconnects, the music would resume seamlessly. Is this possible to do somehow?

The logs:
```
Dec 06 10:17:07 raspberrypi librespot[1321]: [2022-12-06T10:17:07Z WARN  librespot_playback::audio_backend::alsa] Error writing from AlsaSink buffer to PCM, trying to recover, ALSA function 'snd_pcm_writei' failed with error 'EIO: I/O error'
Dec 06 10:17:07 raspberrypi librespot[1321]: [2022-12-06T10:17:07Z ERROR librespot_playback::player] Audio Sink Error On Write: <AlsaSink> ALSA function 'snd_pcm_recover' failed with error 'EIO: I/O error'
Dec 06 10:17:07 raspberrypi systemd[1]: raspotify.service: Main process exited, code=exited, status=1/FAILURE
Dec 06 10:17:07 raspberrypi systemd[1]: raspotify.service: Failed with result 'exit-code'.
Dec 06 10:17:07 raspberrypi systemd[1]: raspotify.service: Consumed 1min 11.030s CPU time.
Dec 06 10:17:17 raspberrypi systemd[1]: raspotify.service: Scheduled restart job, restart counter is at 3.
Dec 06 10:17:17 raspberrypi systemd[1]: Stopped Raspotify (Spotify Connect Client).
Dec 06 10:17:17 raspberrypi systemd[1]: raspotify.service: Consumed 1min 11.030s CPU time.
Dec 06 10:17:17 raspberrypi systemd[1]: Started Raspotify (Spotify Connect Client).
```

**Comment by JasonLG1979** [comment, marked as answer]
No. What's happening is expected behavior IMHO. What you're doing is the equivalent of unplugging a hard drive in the middle of a file transfer. If you did that the file transfer would understandably fail.

The short answer is don't do that if you don't want librespot to crash.
